src/interceptors/fetch/index.ts

```diff
@@ -46,13 +47,21 @@ export class FetchInterceptor extends Interceptor<HttpRequestEventMap> {
 
       this.logger.info('awaiting for the mocked response...')
 
+      const signal = interactiveRequest.signal
+      const rejectWhenRequestAborted = new DeferredPromise<string>()
+
+      signal.addEventListener('abort', () => rejectWhenRequestAborted.reject(signal.reason))
```

Comment:
I think we should implement it the way that user-caused aborts are treated as `AbortError`, as per [spec](https://developer.mozilla.org/en-US/docs/Web/API/AbortController/abort). 

If I'm reading this right, if the user aborts the request, we don't have to do anything about that except stop the request listener promise. If that's the case, then we should instead have a race between two promises that _resolve_. 

But this may be more complicated than that. I'm trying to account for user-caused abort events while working on #398 and I could certainly use your opinion on what's the best way to approach this.

Reply:
We can also use a built-in rejection mechanism `.throwIfAborted` on the signal ([spec](https://dom.spec.whatwg.org/#aborting-ongoing-activities)). 

Perhaps it'd be more semantically correct to do that and handle the thrown `AbortError` as a special case below.

```js
const resolverResult = await until(async () => {
  request.signal.throwIfAborted()

  // the rest of the request listener promises.
})

if (resolverResult.error) {
  if (resolverResult.error instanceof AbortError) {
    // We know request listener lookup halted due to
    // the operation being aborted.
    handleIfNeeded()
    return
  }

  // Handle other exceptions from the request listeners.
}
```

Reply:
I will address your points in the near future but I wanted to let you know that I already have submitted a PR for #398

Reply:
@kettanaito Outside of `msw` realm, when a request is aborted, the request/promise is rejected with a `DOMException` that have `reason` set to `AbortError`. Is there any reason to change this behavior ?

Reply:
I have updated the PR to use the native error instead of a wrapped error.

Reply:
@kettanaito it turns out they do : https://github.com/nodejs/undici/blob/main/test/fetch/abort.js#L78

Reply:
This line of code seems to break for me when I'm not setting an AbortController because signal is `null` in that case.

Reply:
> This line of code seems to break for me when I'm not setting an AbortController because signal is `null` in that case.

When no abort signal is has been provided to the intercepted request, one is injected automatically, so the signal cannot be null.
If you are facing any issue, please open one.